iris_doc/api_tagger.py

```diff
@@ -438,28 +464,44 @@ def _getClassTokens(self, className: str, lineIndex: int, type: str = TYPE_CLASS
             i += 1
         return (classScopeEndIndex, tokens)
 
-    def _getFunctionParameterList(self,
-                                  functionName: str,
-                                  startIndex: int,
-                                  endIndex: int) -> Tuple[List[str], int, int]:
-        parameterList: List[str] = []
-
-        index = startIndex
-        parameterScopeStartIndex = index
-        parameterScopeEndIndex = endIndex
-        while index < endIndex:
+    def _findBoundsMatching(
+            self,
+            function_name: str,
+            start_end: Tuple[int, int],
+            match_start: Callable[[str, str], bool],
+            match_end: Callable[[str], bool]
+    ) -> Tuple[int, int]:
+        index = start_end[0]
+        found_start = False
+        bounds_start_index = index
+        bounds_end_index = start_end[1]
+        while index < start_end[1]:
             line = self.__fileLines[index]
 
-            if self.__syntaxMatcher.matchFunctionParameterScopeStart(functionName, line):
-                parameterScopeStartIndex = index
-            if self.__syntaxMatcher.matchFunctionParameterScopeEnd(line):
-                parameterScopeEndIndex = index
+            if not found_start and match_start(function_name, line):
+                bounds_start_index = index
+                found_start = True
+            if match_end(line):
+                # Because the while loop will never read the last line, this if statement will often not fire.
+                bounds_end_index = index
                 break
 
             index += 1
-        parameterBlockLine = "".join(map(lambda x: x.strip(
-        ), self.__fileLines[parameterScopeStartIndex:parameterScopeEndIndex + 1]))
+        return [bounds_start_index, bounds_end_index]
 
+    def _getFunctionParameterList(self,
+                                  functionName: str,
+                                  startIndex: int,
+                                  endIndex: int) -> Tuple[List[str], int, int]:
+        parameterList: List[str] = []
+        [parameterScopeStartIndex, parameterScopeEndIndex] = self._findBoundsMatching(
+            functionName, [startIndex, endIndex],
+            self.__syntaxMatcher.matchFunctionParameterScopeStart, self.__syntaxMatcher.matchFunctionParameterScopeEnd
+        )
+
+        parameterBlockLine = " ".join(map(
```

Comment:
Can you give more detail why `" "` is needed here, I prefer to not add extra format if possiable.

Reply:
Yes for sure.

Objective C method like look like this:
```objc
- (int)updatePlayerPositionInfo:(NSInteger)playerId
                   positionInfo:(AgoraRemoteVoicePositionInfo* _Nonnull)positionInfo
```

when joined without a space "playerId and "positionInfo" become "playerIdpositionInfo", so it's impossible to know where to separate the parameters

Reply:
@littleGnAl we spoke about this one last week, but let me know if this is alright to go ahead

Reply:
@maxxfrazer I just wondering if we can pass the whole lines of the parameter list to the `findFunctionParameterList`, and move the `join` logic to each language.

Reply:
All set @littleGnAl